agimus_controller_ros/hpp_agimus_controller.py

```diff
@@ -175,28 +170,33 @@ def solve_and_send(self):
             np.save("control_refs.npy", self.control_refs)
         _, u, k = self.mpc.get_mpc_output()
 
+        self.send(sensor_msg, u, k)
+
+    def get_sensor_msg(self):
+        with self.mutex:
+            sensor_msg = deepcopy(self.sensor_msg)
+        return sensor_msg
+
+    def send(self, sensor_msg, u, k):
         self.control_msg.header = Header()
         self.control_msg.header.stamp = rospy.Time.now()
         self.control_msg.feedback_gain = to_multiarray_f64(k)
         self.control_msg.feedforward = to_multiarray_f64(u)
         self.control_msg.initial_state = sensor_msg
         self.control_publisher.publish(self.control_msg)
 
-    def get_sensor_msg(self):
-        with self.mutex:
-            sensor_msg = deepcopy(self.sensor_msg)
-        return sensor_msg
-
     def run(self):
         self.wait_first_sensor_msg()
-        self.plan_and_first_solve()
+        sensor_msg, u, k = self.plan_and_first_solve()
+        input("Press enter to continue ...")
+        self.send(sensor_msg, u, k)
```

Comment:
Personnaly, I would prefer something like 
```suggestion
        input("Press enter to continue ...")
        self.first_solve_and_send()
```
Because we are using `self.send()` in the function where we solve in the `while not rospy.is_shutdown()` loop, or we can let it as it is and also use `self.send()` in the while loop

Reply:
The way you suggest is less user friendly because you expect the robot to move right away when you press enter not wait a bit more that the solver solve the first problem and then move.
So decoupling the `solve` and the `send` part in the init make sens.

For the for loop we will always solve and then send. Decoupling here do not affect the behaviour of the logic.
So if you prefer having them separated here as well it is fine by me.